frontends/web/src/components/headerssync/headerssync.tsx

```diff
@@ -16,89 +16,64 @@
  * limitations under the License.
  */
 
-import { Component} from 'react';
+import { FunctionComponent, useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
 import { CoinCode } from '../../api/account';
-import { subscribe } from '../../decorators/subscribe';
-import { translate, TranslateProps } from '../../decorators/translate';
+import { subscribeCoinHeaders } from '../../api/coins';
+import { useSubscribe } from '../../hooks/api';
 import Spinner from '../spinner/ascii';
 import style from './headerssync.module.css';
 
-interface IHeadersSyncProps {
+interface Props {
     coinCode: CoinCode;
 }
 
-interface IStatus {
-    targetHeight: number;
-    tip: number;
-    tipAtInitTime: number;
-    tipHashHex: string;
+const useMountedRef = () => {
+    const mounted = useRef(true);
+    useEffect(() => (
+        () => {mounted.current = false}
```

Comment:
that does not do anything? or what am i missing?

Reply:
ah this returns the cleanup callback. mistook ( for { above.

better to init to `false` and set to `true` here? `mounted.current` sounds like it should be true while mounted.

Reply:
I think it is alredy doing what you want, the return callback `() => {mounted.current = false}` is called on unmount https://reactjs.org/docs/hooks-effect.html